Backend-tests.md

```diff
@@ -270,17 +245,17 @@ Now consider the following sets of test cases:
 
-This example illustrates something very important about code coverage: **Code coverage less than 100% implies that the tests are not comprehensive, but code coverage of 100% does not imply that tests are comprehensive.** Therefore, while code coverage is a useful tool, you should primarily think about whether your tests cover all the possible behaviors of the code being tested. In other words, you should have a behavior-first perspective. Don't just think about which lines are covered.
+This example illustrates something very important about code coverage: **Code coverage less than 100% implies that the tests are not comprehensive, but code coverage of 100% does NOT imply that tests are comprehensive.** Therefore, while code coverage is a useful tool, you should primarily think about whether your tests cover all the possible behaviors of the code being tested. In other words, you should have a behavior-first perspective. Don't just think about which lines are covered.
 
 ## Write backend tests
 
 ### Backend test structure
 
-We write our backend tests with Python's [unittest framework](https://docs.python.org/3/library/unittest.html).You should familiarize yourself that framework. In particular, you will be using its assertion functions in your tests.
+We write our backend tests with Python's [unittest framework](https://docs.python.org/3/library/unittest.html). You should familiarize yourself that framework by reading through the ["basic example" in its documentation](https://docs.python.org/3/library/unittest.html#basic-example). You should also take a look at what [assertion functions](https://docs.python.org/3/library/unittest.html#unittest.TestCase) are available.
 
-Backend test files live alongside the backend code files they test. For example, alongside `core/controllers/base.py` you'll find `core/controllers/base_test.py`. That `_test.py` suffix is important. It's how we identify files that have tests to run.
+Backend test files live alongside the backend code files they test. For example, alongside `core/controllers/base.py` you'll find `core/controllers/base_test.py`. That `_test.py` suffix is important. It's how we identify which files have tests to run. Note that each file is entirely code or entirely tests. No file mixes code and tests.
 
-Inside test files, tests are organized into classes whose names end in `Tests`. Test cases are methods of these classes, and the method names begin with `test_`. For example, here is an example of a test from `base_test.py`:
+Inside test files, tests are organized into classes whose names end in `Tests`. Test cases are methods of these classes, and the method names begin with `test_`. Note that only methods beginning with `test_` and inside classes ending in `Tests` are executed as tests. Here is an example of a test from `base_test.py`:
```

Comment:
I didn't think the "inside classes ending in `Tests`" was a criterion. Where did it come from? (Did you check manually that this is the case -- i.e. that if a class doesn't end in "Tests" then it isn't executed when calling run_backend_tests.py?)

Note that `test_` is a unittest behaviour (see https://docs.python.org/3/library/unittest.html#basic-example). For the class I thought the eligibility criterion is that it should subclass the relevant test superclass in our codebase.

Reply:
done